### PR TITLE
Correcting array declaration logic

### PR DESCRIPTION
**Pull request by dmikushin**
Make snowman to generate syntactically correct array declarations: 'type varname[size];' instead of wrong 'type[size] varname;'

### CODE REVIEW

**Comment by yegord** [comment]
The patch contains changes unrelated to what is described.
Could you please remove unrelated changes?

**Comment by dmikushin** [comment]
Done!

File: src/nc/core/likec/TreePrinter.cpp [hunk 1]
```diff
@@ -276,7 +276,14 @@ void TreePrinter::printSignature(const FunctionDeclaration *node) {
 }
 
 void TreePrinter::doPrint(const MemberDeclaration *node) {
-    out_ << *node->type() << ' ' << node->identifier() << ';';
+    if (!node->type()->isScalar()) {
```
Comment (yegord) [inline src/nc/core/likec/TreePrinter.cpp]: Could you please make use of nc casts?
Something like:
```
if (auto pointerType = node->type()->as<PointerType>()) {
    if (auto arrayType = pointerType->as<ArrayType>()) {
        // Print array declaration.
        return;
    }
}
// Print the usual declaration.
```
You will not need `reinterpret_cast` then.

Also, isScalar is false not only for array types. It is also false for void and structs, for example. So, it is not very correct to check for array type using isScalar().

File: src/nc/core/likec/Types.h [hunk 2]
```diff
@@ -214,6 +214,8 @@ class ArrayType: public PointerType {
     virtual BitSize sizeOf() const override { return elementType()->size() * length(); }
 
     virtual void print(QTextStream &out) const override;
+
+    virtual bool isScalar() const override { return false; }
```
Comment (yegord) [inline src/nc/core/likec/Types.h]: Just fixed it in master.
Not sure if ArrayType should be PointerType, but let it be for now.

**Comment by yegord** [comment]
> Any chance this can be merged?

No. The patch is incorrect at the moment.

**Review by KostadinDamyanov — APPROVED** [review]

**Pull request closed without merging**